distributed/worker.py

```diff
@@ -2741,18 +2746,20 @@ def get_client(address=None, timeout=None, resolve_address=True) -> Client:
         pass
     else:
         if not address or worker.scheduler.address == address:
-            return worker._get_client(timeout=timeout)
-
-    from distributed.client import Client
+            client = worker._get_client(timeout=timeout)
+            client.asynchronous = asynchronous
+            return client
 
     try:
         client = Client.current()  # TODO: assumes the same scheduler
     except ValueError:
         client = None
+
     if client and (not address or client.scheduler.address == address):
+        client.asynchronous = asynchronous
```

Comment:
In https://github.com/dask/distributed/issues/7658#issue-1627905751 I was even proposing to raise if the type does not match the automatically picked one, otherwise the client will be unusable.

Reply:
Indeed, I was not purposely ignoring your proposal; had it in my back pocket ready to use. When it appeared that it could work w/o it in the current iteration, I naively thought perhaps this was okay. As you point out now, I better understand why raising would be preferred. I will adjust.